### PR TITLE
site healing: Skip stale bucket metadata updates from peer

### PR DESCRIPTION
**Pull request by poornas**
Allow healing to apply bucket metadata change only when peer
gave the most recent update.

## Motivation and Context
Avoid overwriting with stale updates from peer during healing.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 15186-8c0bfd6/gopath/src/github.com/minio
+ git -C 15186-8c0bfd6/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 15186-8c0bfd6/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 15186-8c0bfd6/gopath/src/github.com/minio/minio fetch --quiet minio pull/15186/head:pr15186
+ git -C 15186-8c0bfd6/gopath/src/github.com/minio/minio checkout --quiet pr15186
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/15186-8c0bfd6/gopath
+ make -C 15186-8c0bfd6/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/minio-go/v7 v7.0.30-0.20220623010642-40b13d984ae3
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/minio/pkg v1.1.26
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/minio/sio v0.3.0
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/klauspost/compress v1.15.6
go: downloading github.com/Shopify/sarama v1.30.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/golang-jwt/jwt/v4 v4.4.1
go: downloading github.com/minio/kes v0.19.2
go: downloading github.com/rs/dnscache v0.0.0-20211102005908-e0241e321417
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4
go: downloading github.com/buger/jsonparser v1.1.1
go: downloading golang.org/x/sys v0.0.0-20220502124256-b6088ccd6cba
go: downloading cloud.google.com/go/storage v1.10.0
go: downloading github.com/minio/console v0.19.0
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/bits-and-blooms/bloom/v3 v3.0.1
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/felixge/fgprof v0.9.2
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/minio/dperf v0.4.2
go: downloading github.com/klauspost/readahead v1.4.0
go: downloading github.com/cosnicolaou/pbzip2 v1.0.1
go: downloading github.com/miekg/dns v1.1.48
go: downloading github.com/klauspost/reedsolomon v1.10.0
go: downloading github.com/go-openapi/loads v0.21.1
go: downloading github.com/coreos/go-oidc v2.1.0+incompatible
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/minio/selfupdate v0.4.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/minio/zipindex v0.2.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/philhofer/fwd v1.1.2-0.20210722190033-5c56ac6d0bb9
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/pierrec/lz4 v2.6.1+incompatible
go: downloading github.com/shirou/gopsutil/v3 v3.22.5
go: downloading github.com/tinylib/msgp v1.1.7-0.20211026165309-e818a1881b0e
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/prometheus/client_golang v1.12.1
go: downloading github.com/yargevad/filepathx v1.0.0
go: downloading github.com/zeebo/xxh3 v1.0.0
go: downloading go.etcd.io/etcd/api/v3 v3.5.4
go: downloading go.etcd.io/etcd/client/v3 v3.5.4
go: downloading go.uber.org/atomic v1.9.0
go: downloading golang.org/x/oauth2 v0.0.0-20220411215720-9780585627b5
go: downloading google.golang.org/api v0.78.0
go: downloading go.uber.org/zap v1.21.0
go: downloading github.com/go-ldap/ldap/v3 v3.4.3
go: downloading github.com/coredns/coredns v1.9.2
go: downloading golang.org/x/time v0.0.0-20220224211638-0e9765cccd65
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/minio/simdjson-go v0.4.2
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.5
go: downloading github.com/elastic/go-elasticsearch/v7 v7.17.1
go: downloading github.com/go-sql-driver/mysql v1.6.0
go: downloading github.com/gomodule/redigo v1.8.8
go: downloading github.com/lib/pq v1.10.4
go: downloading github.com/nats-io/nats.go v1.13.1-0.20220308171302-2f2f6968e98d
go: downloading github.com/nats-io/stan.go v0.10.2
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
go: downloading github.com/bits-and-blooms/bitset v1.2.0
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/fraugster/parquet-go v0.10.0
go: downloading github.com/minio/csvparser v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.12
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading github.com/pquerna/cachecontrol v0.1.0
go: downloading gopkg.in/square/go-jose.v2 v2.5.1
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/klauspost/cpuid/v2 v2.0.14
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading go.etcd.io/etcd/client/pkg/v3 v3.5.4
go: downloading google.golang.org/grpc v1.46.0
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading google.golang.org/genproto v0.0.0-20220429170224-98d788798c3e
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/rivo/uniseg v0.2.0
go: downloading github.com/go-openapi/analysis v0.21.2
go: downloading github.com/go-openapi/spec v0.20.5
go: downloading github.com/go-openapi/swag v0.21.1
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/google/pprof v0.0.0-20211214055906-6f57359322fd
go: downloading golang.org/x/net v0.0.0-20220425223048-2871e0cb64e4
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/gorilla/websocket v1.5.0
go: downloading github.com/lestrrat-go/jwx v1.2.19
go: downloading github.com/montanaflynn/stats v0.6.6
go: downloading github.com/fatih/structs v1.1.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20211209120228-48547f28849e
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.4
go: downloading github.com/prometheus/common v0.34.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading google.golang.org/protobuf v1.28.0
go: downloading github.com/golang/snappy v0.0.4
go: downloading github.com/coreos/go-systemd/v22 v22.3.2
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading cloud.google.com/go v0.100.2
go: downloading go.uber.org/multierr v1.8.0
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/rs/xid v1.4.0
go: downloading gopkg.in/ini.v1 v1.66.4
go: downloading github.com/nats-io/nkeys v0.3.0
go: downloading github.com/minio/md5-simd v1.1.2
go: downloading github.com/apache/thrift v0.15.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20201227073835-cf1acfcdf475
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
go: downloading github.com/go-openapi/strfmt v0.21.2
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/lestrrat-go/backoff/v2 v2.0.8
go: downloading github.com/lestrrat-go/option v1.0.0
go: downloading github.com/lestrrat-go/iter v1.0.1
go: downloading github.com/lestrrat-go/blackmagic v1.0.0
go: downloading github.com/lestrrat-go/httpcc v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/go-openapi/jsonreference v0.20.0
go: downloading github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
go: downloading github.com/mitchellh/mapstructure v1.4.3
go: downloading go.mongodb.org/mongo-driver v1.8.4
go: downloading github.com/go-openapi/errors v0.20.2
go: downloading github.com/oklog/ulid v1.3.1
go: downloading cloud.google.com/go/iam v0.2.0
go: downloading github.com/googleapis/gax-go/v2 v2.3.0
go: downloading go.opencensus.io v0.23.0
go: downloading golang.org/x/xerrors v0.0.0-20220411194840-2f41105eb62f
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/go-stack/stack v1.8.1
go: downloading github.com/go-openapi/runtime v0.23.3
go: downloading github.com/go-openapi/validate v0.21.0
go: downloading github.com/minio/mc v0.0.0-20220512134321-aa60a8db1e4d
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/unrolled/secure v1.10.0
go: downloading github.com/tidwall/gjson v1.14.0
go: downloading github.com/tidwall/pretty v1.2.0
go: downloading github.com/tidwall/match v1.1.1
go: downloading github.com/docker/go-units v0.4.0
go: downloading github.com/gdamore/tcell/v2 v2.4.1-0.20210905002822-f057f0a857a1
go: downloading github.com/pkg/xattr v0.4.5
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/minio/colorjson v1.0.2
go: downloading github.com/charmbracelet/lipgloss v0.5.0
go: downloading github.com/charmbracelet/bubbletea v0.20.0
go: downloading github.com/rivo/tview v0.0.0-20220216162559-96063d6082f3
go: downloading github.com/navidys/tvxwidgets v0.1.0
go: downloading maze.io/x/duration v0.0.0-20160924141736-faac084b6075
go: downloading github.com/charmbracelet/bubbles v0.10.3
go: downloading github.com/posener/complete v1.2.3
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/briandowns/spinner v1.18.1
go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211
go: downloading github.com/prometheus/prom2json v1.3.1
go: downloading github.com/minio/filepath v1.0.0
go: downloading gopkg.in/h2non/filetype.v1 v1.0.5
go: downloading github.com/tklauser/go-sysconf v0.3.10
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/muesli/termenv v0.11.1-0.20220212125758-44cd13922739
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/muesli/ansi v0.0.0-20211031195517-c9f0611b6c70
go: downloading github.com/containerd/console v1.0.3
go: downloading github.com/tklauser/numcpus v0.4.0
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/gdamore/encoding v1.0.0
go: downloading cloud.google.com/go/compute v1.6.1
../../../../pkg/mod/github.com/minio/console@v0.19.0/pkg/logger/config/config.go:20:2: github.com/minio/madmin-go@v1.3.14: replacement directory ../madmin-go does not exist
make: *** [build] Error 1

```

File: go.mod [hunk 1]
```diff
@@ -222,3 +222,4 @@ require (
 	gopkg.in/square/go-jose.v2 v2.5.1 // indirect
 	maze.io/x/duration v0.0.0-20160924141736-faac084b6075 // indirect
 )
+replace github.com/minio/madmin-go => ../madmin-go
```
Comment (harshavardhana) [inline go.mod]: madmin-go changes are in and tagged @poornas

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (poornas) [reply go.mod]: thanks, updated

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Please add a test @poornas using available shell scripts.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#a7705afa-320f-4600-bcc8-d9968e7f56be) |
| mint-pools.sh | :x: [more...](#09a01945-dd80-47a3-956f-43804264ae9c) |
<a name="a7705afa-320f-4600-bcc8-d9968e7f56be"></a>
### 15186-8433a37/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:30879
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 56 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 31 seconds
FAILED in 3 minutes and 38 seconds
Unhandled exception. System.AggregateException: One or more errors occurred. (A task was canceled.)
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Threading.Tasks.Task.GetExceptions(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/Program.cs:line 155
--- End of stack trace from previous location ---

   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/Program.cs:line 155
(8/15) Running minio-go tests ... done in 1 minutes and 21 seconds
(9/15) Running minio-java tests ... done in 32 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 5 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 18 seconds
(12/15) Running s3cmd tests ... done in 25 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 14 out of 15 tests successfully.
```
<a name="09a01945-dd80-47a3-956f-43804264ae9c"></a>
### 15186-8433a37/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30609
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 7 seconds
(5/15) Running awscli tests ... done in 55 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 34 seconds
FAILED in 3 minutes and 29 seconds
{
  "name": "minio-dotnet : ListObjects_Test6",
  "function": "IObservable<Item> ListObjectsAsync(string bucketName, string prefix = null, bool recursive = false, CancellationToken cancellationToken = default(CancellationToken))",
  "args": {
    "bucketName": "minio-dotnet-example-svsdjdcxa2d2gjy",
    "objectName": "minio-dotnet-example-pb6d2ifw23",
    "recursive": "false"
  },
  "duration": 80627,
  "status": "FAIL",
  "alert": "MinIO API responded with message=Connection error:An error occurred while sending the request.. Status code=0, response=An error occurred while sending the request., content=",
  "message": "Minio.Exceptions.ConnectionException: MinIO API responded with message=Connection error:An error occurred while sending the request.. Status code=0, response=An error occurred while sending the request., content=\n   at Minio.MinioClient.ParseErrorNoContent(ResponseResult response) in /mint/run/core/minio-dotnet/temp/Minio/MinioClient.cs:line 581\n   at Minio.MinioClient.ParseError(ResponseResult response) in /mint/run/core/minio-dotnet/temp/Minio/MinioClient.cs:line 564\n   at Minio.MinioClient.<>c.<.ctor>b__130_0(ResponseResult response) in /mint/run/core/minio-dotnet/temp/Minio/MinioClient.cs:line 54\n   at Minio.MinioClient.HandleIfErrorResponse(ResponseResult response, IEnumerable`1 handlers, DateTime startTime) in /mint/run/core/minio-dotnet/temp/Minio/MinioClient.cs:line 761\n   at Minio.MinioClient.ExecuteTaskCoreAsync(IEnumerable`1 errorHandlers, HttpRequestMessageBuilder requestMessageBuilder, CancellationToken cancellationToken, Boolean isSts) in /mint/run/core/minio-dotnet/temp/Minio/MinioClient.cs:line 542\n   at Minio.MinioClient.PutObjectSinglePartAsync(PutObjectArgs args, CancellationToken cancellationToken) in /mint/run/core/minio-dotnet/temp/Minio/ApiEndpoints/ObjectOperations.cs:line 1162\n   at Minio.MinioClient.PutObjectAsync(PutObjectArgs args, CancellationToken cancellationToken) in /mint/run/core/minio-dotnet/temp/Minio/ApiEndpoints/ObjectOperations.cs:line 655\n   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/FunctionalTest.cs:line 865\n   at Minio.Functional.Tests.FunctionalTest.ListObjects_Test6(MinioClient minio) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/FunctionalTest.cs:line 4664"
}
(8/15) Running minio-go tests ... done in 1 minutes and 27 seconds
(9/15) Running minio-java tests ... done in 38 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 4 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 36 seconds
(12/15) Running s3cmd tests ... done in 24 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
@poornas PTAL at the conflicts.

**Review by harshavardhana — APPROVED** [review]

File: docs/site-replication/run-multi-site-ldap.sh [hunk 2]
```diff
@@ -237,4 +237,31 @@ if [ "${enabled_minio1}" != "Enabled" ]; then
...
+fi
+# stop minio1
+ps aux  |  grep -i '[m]inio-ldap-idp1'  |  awk '{print $2}'  |  xargs sudo kill -9
+# Update tag on minio2/newbucket when minio1 is down
+./mc tag set minio2/newbucket "key=val2"
+# Restart minio1 instance
+minio server --config-dir /tmp/minio-ldap --address ":9001" /tmp/minio-ldap-idp1/{1...4} >/tmp/minio1_1.log 2>&1 &
+sleep 10
+# Test whether most recent tag update on minio2 is replicated to minio1
+val=$(mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```
Comment (harshavardhana) [inline docs/site-replication/run-multi-site-ldap.sh]:
```suggestion
val=$(./mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```

File: docs/site-replication/run-multi-site-minio-idp.sh [hunk 3]
```diff
@@ -268,3 +268,30 @@ if [ "${enabled_minio1}" != "Enabled" ]; then
...
+fi
+
+ps aux  |  grep -i '[m]inio-internal-idp1'  |  awk '{print $2}'  |  xargs sudo kill -9
+# Update tag on minio2/newbucket when minio1 is down
+./mc tag set minio2/newbucket "key=val2"
+# Restart minio1 instance
+minio server --config-dir /tmp/minio-internal --address ":9001" /tmp/minio-internal-idp1/{1...4} >/tmp/minio1_1.log 2>&1 &
+sleep 10
+# Test whether most recent tag update on minio2 is replicated to minio1
+val=$(mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```
Comment (harshavardhana) [inline docs/site-replication/run-multi-site-minio-idp.sh]:
```suggestion
val=$(./mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```

File: docs/site-replication/run-multi-site-oidc.sh [hunk 4]
```diff
@@ -222,3 +222,30 @@ if [ "${enabled_minio1}" != "Enabled" ]; then
...
+fi
+# stop minio1 instance
+ps aux  |  grep -i '[m]inio1'  |  awk '{print $2}'  |  xargs sudo kill -9
+# Update tag on minio2/newbucket when minio1 is down
+./mc tag set minio2/newbucket "key=val2"
+# Restart minio1 instance
+minio server --address ":9001" --console-address ":10000" /tmp/minio1/{1...4} >/tmp/minio1_1.log 2>&1 &
+sleep 10
+# Test whether most recent tag update on minio2 is replicated to minio1
+val=$(mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```
Comment (harshavardhana) [inline docs/site-replication/run-multi-site-oidc.sh]:
```suggestion
val=$(./mc tag list minio1/newbucket --json | jq -r .tagset | jq -r .key )
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Some suggestions instead of relying on `ps aux` to capture the pid's make sure to capture whatever you want based on 

```
minio server --config-dir /tmp/minio-ldap --address ":9001" /tmp/minio-ldap-idp1/{1...4} >/tmp/minio1_1.log 2>&1 &
site1_pid=$!

minio server --config-dir /tmp/minio-ldap --address ":9002" /tmp/minio-ldap-idp2/{1...4} >/tmp/minio2_1.log 2>&1 &
site2_pid=$!

minio server --config-dir /tmp/minio-ldap --address ":9003" /tmp/minio-ldap-idp3/{1...4} >/tmp/minio3_1.log 2>&1 &
site3_pid=$!
```

Then you can use this to kill the relevant PID later on.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#0ff2caaf-af69-49c3-84d4-13d6750c0cfc) |
| mint-dist-erasure.sh | :x: [more...](#e03be7fe-efda-4689-ad08-bbac41ba66b3) |
<a name="e03be7fe-efda-4689-ad08-bbac41ba66b3"></a>
### 15186-0f6390b/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:31394
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 57 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 46 seconds
{
  "name": "mc",
  "duration": "40127",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-15694 --quiet --no-color cp /mint/data/datafile-65-MB myminio/mc-test-bucket-3801/mc-test-object-18405  >>> \n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-3801/mc-test-object-18405`\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Get \"http://minio-k8s1:31394/mc-test-bucket-3801/?location=\": dial tcp 15.15.15.1:31394: i/o timeout\nTotal: 0 B, Transferred: 0 B, Speed: 0 B/s"
}
(7/15) Running minio-dotnet tests ... done in 6 minutes and 29 seconds
(8/15) Running minio-go tests ... done in 1 minutes and 21 seconds
(9/15) Running minio-java tests ... done in 28 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 3 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 20 seconds
(12/15) Running s3cmd tests ... done in 24 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 14 out of 15 tests successfully.
```
<a name="0ff2caaf-af69-49c3-84d4-13d6750c0cfc"></a>
### 15186-0f6390b/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31382
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 54 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 22 seconds
{
  "name": "mc",
  "duration": "10418",
  "function": "test_cat_object",
  "status": "FAIL",
  "error": "unable to download object using 'mc cat'"
}
(7/15) Running minio-dotnet tests ... done in 6 minutes and 59 seconds
(8/15) Running minio-go tests ... done in 1 minutes and 3 seconds
(9/15) Running minio-java tests ... done in 19 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 1 seconds
(11/15) Running minio-py tests ... done in 47 seconds
(12/15) Running s3cmd tests ... done in 23 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running versioning tests ... done in 3 minutes and 1 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**